jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/hive/HiveColumnMetadataReader.java

```diff
@@ -0,0 +1,66 @@
+package com.exasol.adapter.dialects.hive;
+
+import com.exasol.adapter.*;
+import com.exasol.adapter.dialects.*;
+import com.exasol.adapter.jdbc.*;
+import com.exasol.adapter.metadata.*;
+
+import java.sql.*;
+import java.util.regex.*;
+
+import static com.exasol.adapter.dialects.hive.HiveProperties.HIVE_CAST_NUMBER_TO_DECIMAL_PROPERTY;
+
+/**
+ * This class implements Hive-specific reading of column metadata.
+ */
+public class HiveColumnMetadataReader extends BaseColumnMetadataReader {
+
+    /**
+     * Create a new instance of the {@link HiveColumnMetadataReader}
+     *
+     * @param connection          connection to the remote data source
+     * @param properties          user-defined adapter properties
+     * @param identifierConverter converter between source and Exasol identifiers
+     */
+    public HiveColumnMetadataReader(final Connection connection, final AdapterProperties properties,
+            final IdentifierConverter identifierConverter) {
+        super(connection, properties, identifierConverter);
+    }
+
+    @Override
+    public DataType mapJdbcType(final JdbcTypeDescription jdbcTypeDescription) {
+        switch (jdbcTypeDescription.getJdbcType()) {
+        case Types.DECIMAL:
+            return mapDecimal(jdbcTypeDescription);
+        default:
+            return super.mapJdbcType(jdbcTypeDescription);
+        }
+    }
+
+    protected DataType mapDecimal(final JdbcTypeDescription jdbcTypeDescription) {
```

Comment:
This is duplication with the `OracleColumnMetadataReader`. Please move the common part to `BaseColumnMetadataReader` as `protected` methods.

Reply:
Duplication removed